src/platform/nm-platform-utils.c

```diff
@@ -404,7 +440,31 @@ nmp_utils_ethtool_set_link_settings (int ifindex,
 	edata.cmd = ETHTOOL_SSET;
 	if (autoneg) {
 		edata.autoneg = AUTONEG_ENABLE;
-		edata.advertising = edata.supported;
+		if (!speed)
+			edata.advertising = edata.supported;
+		else {
+			guint32 mode;
+
+			mode = get_baset_mode (speed, duplex);
+
+			if (!mode) {
+				nm_log_trace (LOGD_PLATFORM,
+				              "ethtool[%d]: %uBASE-T %s duplex mode cannot be advertised",
+				              ifindex,
+				              speed,
+				              (duplex == NM_PLATFORM_LINK_DUPLEX_HALF) ? "Half" : "Full");
+				return FALSE;
+			}
+			if (!(edata.supported & mode)) {
+				nm_log_trace (LOGD_PLATFORM,
+				              "ethtool[%d]: device does not support %uBASE-T %s duplex mode",
+				              ifindex,
+				              speed,
+			                      (duplex == NM_PLATFORM_LINK_DUPLEX_HALF) ? "Half" : "Full");
```

Comment:
indentation. Also, `link_duplex_to_string()` returns lower-case words (not "Half" and "Full"). Can you move `link_duplex_to_string()` to nm-platform.c, and re-use it?

Reply:
Done